### PR TITLE
xtask: add no_infer_bin_path flag

### PR DESCRIPTION
**Pull request by gabriele-0201**
remove the ci flag and set the default behavior to add binaries to the path and CONSTANTS_MANIFEST to environment variables

### CODE REVIEW

**Comment by gabriele-0201** [comment]
Current dependencies on/for this PR:
* **#252** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/252?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/thrumdev/blobs/252?utm_source=stack-comment).

File: xtask/src/cli.rs [hunk 1]
```diff
@@ -57,9 +57,11 @@ pub mod test {
 
     #[derive(Debug, Args)]
     pub struct Params {
-        /// If the test is executed in CI
+        /// By default, binary paths are extracted using the `cargo metadata` command under the key `target_directory`.
```
Comment (pepyakin) [inline xtask/src/cli.rs:60]: Nit: paths to compiled binaries

Comment (pepyakin) [reply xtask/src/cli.rs:60]: Nit: no need details about target I suppose

**Review by pepyakin — APPROVED** [review]
This looks good, way better than prior proposals.

However, it still modifies the env vars of the process. This is fine for now, just a bit sketchy. A cleaner approach would be to supply env variables to each shell invocation.

**Pull request merged**